### PR TITLE
Add ECS and SNS modules

### PR DESCRIPTION
**Pull request by jgchoppe**

### CODE REVIEW

File: tf/2-variables.tf [hunk 1]
```diff
@@ -20,6 +20,12 @@ variable "tags" {
   default     = {}
 }
 
+variable "account_id" {
```
Comment (simonmeyerrr) [inline tf/2-variables.tf:23]: You can retrieve your account id with `aws_caller_identity`

File: tf/ecs.tf [hunk 2]
```diff
@@ -0,0 +1,310 @@
+resource "aws_ecs_cluster" "cluster" {
+  name = var.ecs_cluster_name
+}
+
+# https://github.com/cloudposse/terraform-aws-ecs-container-definition
```
Comment (simonmeyerrr) [inline tf/ecs.tf:5]: Use the link with version tag please

File: tf/ecs.tf [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+
+  container_memory = var.ecs_container_memory
+  container_cpu    = var.ecs_task_cpu
+}
+
+resource "aws_ecs_task_definition" "task" {
+  container_definitions = module.ecs-container-definition.json_map_encoded_list
+  family                = var.ecs_task_family
+
+  # TaskRoleArn
```
Comment (simonmeyerrr) [inline tf/ecs.tf:38]: Please use `##` for documentation comment

File: tf/ecs.tf [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+resource "aws_iam_role_policy_attachment" "execution_role_attached_policy" {
+  count      = length(var.ecs_execution_role_policies)
+  policy_arn = var.ecs_execution_role_policies[count.index]
+  role       = aws_iam_role.ecs_execution_role.name
+}
+
+# AWS IAM Role and Policy for cloudwatch
+resource "aws_iam_role" "scheduled_task_cloudwatch" {
+  name               = "ecs-cloudwatch-role"
+  assume_role_policy = <<EOF
```
Comment (simonmeyerrr) [inline tf/ecs.tf:267]: Use `jsonencode`

File: tf/ecs.tf [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy" "scheduled_task_cloudwatch_policy" {
+  name   = "ecs-cloudwatch-policy"
+  role   = aws_iam_role.scheduled_task_cloudwatch.id
+  policy = <<EOF
```
Comment (simonmeyerrr) [inline tf/ecs.tf:287]: Use `jsonencode`

File: tf/ecs.tf [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+      "Effect": "Allow",
+      "Action": "iam:PassRole",
+      "Resource": [
+        "*"
+      ]
+    }
+  ]
+}
+EOF
+}
```
Comment (simonmeyerrr) [inline tf/ecs.tf:310]: Missing end of line

File: tf/sns.tf [hunk 7]
```diff
@@ -0,0 +1,5 @@
+# https://registry.terraform.io/modules/terraform-aws-modules/sns/aws/2.1.0
+
+resource "aws_sns_topic" "sns_topic" {
+  name = var.sns_name
+}
```
Comment (simonmeyerrr) [inline tf/sns.tf:5]: Missing end of line

**Review by simonmeyerrr — CHANGES_REQUESTED** [review]
Please, make sure to add `tags` attribute on every compatible resources.
Use `##` for documentation comment.

File: envs/qa.tfvars [hunk 8]
```diff
@@ -18,6 +19,16 @@ cognito_client_name = "terraform-boilerplate"
...
+// -- ECS
+ecs_role_name = "ecs-access-boilerplate"
+ecs_container_memory = 2048
+ecs_task_family = "ecs-task-boilerplate"
+ecs_container_name = "app"
+ecs_execution_role_name = "BoilerplateECSExecutionRole"
+ecs_cluster_name = "ecs-cluster-boilerplate"
+ecs_service_name = "ecs-service-boilerplate"
+ecs_schedule_expression = "rate(7 days)"
+enable_ecs_scheduling = true
```
Comment (jklakosz) [inline envs/qa.tfvars:30]: rename it `ecs_enable_scheduling`

**Review by jklakosz — CHANGES_REQUESTED** [review]

**Review by simonmeyerrr — APPROVED** [review]

**Review by jklakosz — APPROVED** [review]

**Pull request merged**